test/helpers/generate-matrix.test.ts

```diff
@@ -36,4 +36,70 @@ describe('generateMatrix', () => {
       ]
     });
   });
+
+  it('should generate matrix json with appropriate load balancing in naive case', () => {
```

Comment:
Might be good to explain what naive, moderate, and advanced cases are

Reply:
a better demarcation would be `small | medium | large`